onsite/phosphors/phosphors.py

```
@@ -727,17 +739,60 @@ def get_intensity_weight(intensity):
+        spec_gen.getSpectrum(spec, seq, 1, max_z)
     except Exception:
         return []
-    if spec.size() == 0:
+    peaks = spec.get_peaks()
+    if not peaks or len(peaks[0]) == 0:
         return []
-    return sorted(float(m) for m in spec.get_peaks()[0])
+    mzs = peaks[0]
+    try:
+        annotations = spec.getStringDataArrays()[0]
+    except Exception:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)